scripts/quests/otherAreas/For_the_Birds.lua

```diff
@@ -0,0 +1,211 @@
+                    npcUtil.giveKeyItem(player, xi.ki.GLITTERING_FRAGMENT)
+                    quest:setVar(player, 'Prog', 2)
+                end,
+
+                [18] = function(player, csid, option, npc)
+                    if quest:complete(player) then
+                        xi.quest.setMustZone(player, xi.questLog.OTHER_AREAS, xi.quest.id.otherAreas.BETTER_THE_DEMON_YOU_KNOW) -- must zone prior to For the Birds
+                    end
+                end,
+            },
+        },
+
+        [xi.zone.CASTLE_OZTROJA] =
+        {
+            ['Daa_Bola_the_Seer'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 0 then
+                        return quest:event(86)
+                    else
+                        return quest:messageName(oztrojaID.text.LETTING_YOU_GO)
+                    end
+                end,
+
+                onTrade = function(player, npc, trade)
+                    if
+                        quest:getVar(player, 'Prog') == 0 and
+                        npcUtil.tradeHasExactly(trade, { xi.item.ARNICA_ROOT })
+                    then
+                        return quest:progressEvent(87, { [1] = xi.item.ARNICA_ROOT })
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                [87] = function(player, csid, option, npc)
+                    player:tradeComplete()
+                    quest:setVar(player, 'Prog', 1)
+                end,
+            },
+        },
+
+        [xi.zone.BEADEAUX] =
+        {
+            ['GeFhu_Yagudoeye'] =
+            {
+                onTrigger = function(player, npc)
+                    if
+                        quest:getVar(player, 'Prog') == 2 and
+                        player:hasKeyItem(xi.ki.GLITTERING_FRAGMENT) and
+                        npcUtil.popFromQM(player, npc, mobListTable, { claim = true, hide = 0, })
+                    then
+                        player:delKeyItem(xi.ki.GLITTERING_FRAGMENT)
+                        player:messageSpecial(beadeauxID.text.GLITTERING_FRAGMENT_STOLEN, 0, xi.ki.GLITTERING_FRAGMENT)
+                        return player:messageSpecial(beadeauxID.text.THE_QUADAV_ARE_ATTACKING)
+                    elseif quest:getVar(player, 'Prog') == 3 then
+                        return quest:progressEvent(124)
+                    elseif quest:getVar(player, 'Prog') == 4 then
+                        return quest:messageName(beadeauxID.text.HU_URR)
+                    end
+                end,
+            },
+
+            ['Nickel_Quadav'] =
+            {
+                onMobDeath = function(mob, player, optParams)
+                    if
+                        quest:getVar(player, 'Prog') == 2 and
+                        allDead()
+                    then
+                        quest:setVar(player, 'Prog', 3)
+                    end
```

Comment:
```lua
                    if
                        quest:getVar(player, 'Prog') == 2 and
                        not player:hasKeyItem(xi.ki.GLITTERING_FRAGMENT) and
                        allDead()
                    then
                        quest:setVar(player, 'Prog', 3)
                    end
```
This can still be cheesed. We may need to set a local var on the mobs on pop with the player ID of the one who poped them so we can check it on mob death aswell, so only that player that poped them can progress the quest.

What we dont want:
- 2 players in party with KIs. Player 1 pops and they fail. Player 1 has no KI.
- Player 2 pops. Since player 1 doesnt have the KI, he would still get credit even though he wasnt the poper this time.

Reply:
Maybe something like this?

```
local function assignPlayer()
    for mobId = beadeauxID.mob.MAGNES_QUADAV, beadeauxID.mob.MAGNES_QUADAV + 3 do
        local mob = GetMobByID(mobId)
        mob:setLocalVar('PLAYERID', player:getID())
    end
end
```

Reply:
```
if
    quest:getVar(player, 'Prog') == 2 and
    allDead() and
    not player:hasKeyItem(xi.ki.GLITTERING_FRAGMENT) and
    mob:getLocalVar('PLAYERID') == player:getID()
then
    quest:setVar(player, 'Prog', 3)
end
```